src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
+        if (size == elementData.length) {
+            elementData = expand(elementData);
+        }
 
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Incorrect index " + index + " for ArrayList size " + size);
```
![image](https://github.com/mate-academy/jv-array-list/assets/20326858/2ce8be08-dce1-4c65-9977-923e5b43a4d6)

Reply:
not fixed